### PR TITLE
nightmare vision goggles give you mare vision

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## About The Pull Request

Daniel: https://github.com/tgstation/tgstation/pull/76600
The cooler daniel:
image

## Why It's Good For The Game

Neigh
image

## Changelog

:cl:
add: Nightmare vision goggles now give you mare vision
/:cl:

### CODE REVIEW

**Comment by Jolly-66** [comment]
no way the virtual boy

**Comment by Fikou** [comment]
neeigghhhh

**Comment by nukularpower** [comment]
plz replace psyker vision with this

File: code/modules/clothing/glasses/_glasses.dm [hunk 1]
```diff
@@ -679,6 +679,31 @@
 	inhand_icon_state = "glasses"
 	glass_colour_type = /datum/client_colour/glass_colour/nightmare
 	forced_glass_color = TRUE
+	var/datum/hallucination/stored_hallucination
```
Comment (ArcaneMusic) [inline code/modules/clothing/glasses/_glasses.dm:682]: Autodoc var?

**Review by ArcaneMusic — CHANGES_REQUESTED** [review]

File: code/modules/clothing/glasses/_glasses.dm (hunk 1 above)
Comment (ChungusGamer666) [reply code/modules/clothing/glasses/_glasses.dm:682]: ok whatever

**Comment by san7890** [comment]
this is funny but i feel like you lose out on the whole "WOAH THAT'S MARES" with the complete red/black bullshittery going on. ultimately whatever since it's a meme but it's really hard to tell that's meant to be a mare with the bichromal vision

File: code/modules/clothing/glasses/_glasses.dm [hunk 2]
```diff
@@ -679,6 +679,32 @@
 	inhand_icon_state = "glasses"
 	glass_colour_type = /datum/client_colour/glass_colour/nightmare
 	forced_glass_color = TRUE
+	/// Hallucination datum currently being used for seeing mares
+	var/datum/hallucination/stored_hallucination
+
+/obj/item/clothing/glasses/nightmare_vision/Destroy()
+	if(stored_hallucination)
+		QDEL_NULL(stored_hallucination)
```
Comment (MrMelbert) [inline code/modules/clothing/glasses/_glasses.dm:687]:
```suggestion
	QDEL_NULL(stored_hallucination)
```

File: code/modules/clothing/glasses/_glasses.dm [hunk 3]
```diff
@@ -679,6 +679,32 @@
...
+		affects_us = TRUE, \
+		affects_others = TRUE, \
+		skip_nearby = FALSE, \
+		play_wabbajack = FALSE, \
+	)
+
+/obj/item/clothing/glasses/nightmare_vision/dropped(mob/living/user)
+	. = ..()
+	if(stored_hallucination)
+		QDEL_NULL(stored_hallucination)
```
Comment (MrMelbert) [inline code/modules/clothing/glasses/_glasses.dm:707]:
```suggestion
	QDEL_NULL(stored_hallucination)
```

**Review by MrMelbert — COMMENTED** [review]
It might be better to make this like a 1/50 chance when you equip the goggles rather than every time, just enough to make you question your sanity

**Comment by ChungusGamer666** [comment]
> It might be better to make this like a 1/50 chance when you equip the goggles rather than every time, just enough to make you question your sanity

Will do

File: code/modules/clothing/glasses/_glasses.dm [hunk 4]
```diff
@@ -680,6 +680,32 @@
...
+
+/obj/item/clothing/glasses/nightmare_vision/equipped(mob/living/user, slot)
+	. = ..()
+	if(!(slot & ITEM_SLOT_EYES))
+		return
+	//5% chance to get mare vision
+	if(prob(5))
+		stored_hallucination = 	user.cause_hallucination( \
+			/datum/hallucination/delusion/preset/mare, \
+			"[src]", \
```
Comment (san7890) [inline code/modules/clothing/glasses/_glasses.dm:698]:
```suggestion
			src.name, \
```

we could also just do `name` but this is a bit more explicit...

**Pull request merged**